libpromises/evalfunction.c

```diff
@@ -1188,6 +1188,65 @@ static FnCallResult FnCallGetGid(ARG_UNUSED EvalContext *ctx, ARG_UNUSED const P
 #endif
 }
 
+/*********************************************************************/ 
+
+static FnCallResult FnCallUserInGroup(ARG_UNUSED EvalContext *ctx, ARG_UNUSED const Policy *policy, const FnCall *fp, const Rlist *finalargs)
+{
+    assert(fp != NULL);
+#ifdef _WIN32
+    Log(LOG_LEVEL_ERR, "Function '%s' is POSIX specific", fp->name);
+    return FnFailure();  
+#else
+    assert(finalargs != NULL);
+
+    const char *user_name = RlistScalarValue(finalargs);
+    const char *group_name = RlistScalarValue(finalargs->next);
+
+    StringSet *groups = StringSetNew();
+    
+    errno = 0;
+    struct group *grent = getgrnam(group_name);
+
+    if (errno != 0)
+    {
+        char *error_msg = GetErrorStr()
+        Log(LOG_LEVEL_ERR, "Couldn't open group database in function '%s': %s", fp->name, error_msg);
+        StringSetDestroy(groups);
+        return FnFailure();
+    }
+
+    if (grent == NULL) {
+        StringSetDestroy(groups);
+        return FnReturnContext(false);
+    }
+    else 
+    {
+        switch(errno)
+        {
+            case ENOENT:
+            case ESRCH:
+            case EBADF:
+            case EPERM:
+                StringSetDestroy(groups);
+                return FnReturnContext(false);
+            default:
+                break;
+        }
```

Comment:
This code will never be run, because `errno` is only set when `getgrnam()` returns `NULL`